tests/instances.tcl

```diff
@@ -92,10 +91,30 @@ proc spawn_instance {type base_port count {conf {}}} {
         close $cfg
 
         # Finally exec it and remember the pid for later cleanup.
-        set pid [exec_instance $type $cfgfile]
-        lappend ::pids $pid
+        set retry 100
+        while {$retry} {
+            set pid [exec_instance $type $cfgfile]
+            after 1000
```

Comment:
why is this sleep needed?
original code didn't need a sleep between `exec_instance` and `server_is_up` (which does the necessary sleep and avoids an excessive one).

Reply:
Under normal conditions, the sleep is not needed. But when the system is unstable, some time is needed to avoid misjudging a port not available.

Reply:
@fayadexinqing first this long sleep will dramatically slow down the test suite (it isn't only executed on the rare collision, but every single time).

secondly, i don't see why we need a fixed sleep. there's already a series of sleeps in `server_is_up`, the whole idea is that it probes the server in a tight (50ms) interval so that as soon as it's up we can continue, and we don't have any excessive sleep.

please remove the sleep, re-test the patch in the busy environment that reproduces it and confirm that it is not needed.

if you do see problems, might you might need instead is to change the `retrynum` argument for `server_is_up` from 100 to 120 (or more), this adds another second, in case 5 seconds timeout (100*50) is not enough on your busy system.

Reply:
As you said, a fixed sleep here is duplicated with server_is_up. I re-test the patch after remove it in the unstable system several times, the test go well.

Reply:
I'm not sure i understand your last post. did you test it already or going to test?
Waiting for your test before i merge.